tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -22,6 +99,63 @@ public class CoreclrTestWrapperLib
         public const int DEFAULT_TIMEOUT = 1000 * 60*10;
         public const string GC_STRESS_LEVEL = "__GCSTRESSLEVEL";
 
+        public const string COLLECT_DUMPS_ENVIRONMENT_VAR = "__CollectDumps";
+        public const string CRASH_DUMP_FOLDER_ENVIRONMENT_VAR = "__CrashDumpFolder";
+
+        static bool CollectCrashDump(Process process, string path)
+        {
+            using (var crashDump = File.OpenWrite(path))
+            {
+                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+            }
+        }
+
+        static unsafe bool TryFindChildProcessByName(Process process, string childName, out Process child)
+        {
+            IntPtr snapshot = Kernel32.CreateToolhelp32Snapshot(Kernel32.Toolhelp32Flags.TH32CS_SNAPPROCESS, 0);
```

Comment:
We could use a `SafeHandle`; it just requires implementing a new derived class of the appropriate parent (in this case `SafeHandleMinusOneIsInvalid`).